### PR TITLE
`lang/c`: Add simple HTTP program

### PR DESCRIPTION
**Pull request by razvand**
Add source code, Makefile, prebuilt ELF, pre-etracted dynamic libraries. Document steps to (re)build and run in `README.md`.

### CODE REVIEW

File: lang/c/http_server/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+```text
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN"><html><head><title>It works!</title></head><body><h1>It works!</h1><p>This is only a test.</p></body></html>
+```
+
+## Run with Unikraft
+
+It's easiest to run the server with the binary-compatibility mode of Unikraft by using [the `run.sh` script in the `run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader/blob/master/run.sh):
+
+```console
+sudo ./run.sh -n -r ../dynamic-apps/lang/c++/http_server /http_server
```
Comment (StefanJum) [inline lang/c/http_server/README.md:63]:
```suggestion
sudo ./run.sh -n -r ../dynamic-apps/lang/c/http_server /http_server
```

File: lang/c/http_server/http_server.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
```
Comment (StefanJum) [inline lang/c/http_server/http_server.c:32]: I think this is the outdated licence header right?

**Review by StefanJum — CHANGES_REQUESTED** [review]
Few comments.
Besides those I would move the `helloworld`, `client` and `server` files from `lang/c/` to `lang/c/helloworld/`, `lang/c/server-client` or something similar. Right now the `http_server` directory gets lost between the bunch of files under `lang/c/`.

File: lang/c/http_server/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# Simple C HTP Server
```
Comment (StefanJum) [inline lang/c/http_server/README.md:1]:
```suggestion
# Simple C HTTP Server
```

**Comment by razvand** [comment]
@StefanJum , see now.

**Review by StefanJum — APPROVED** [review]
All good now.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request merged**